Common/src/main/java/com/techarium/techarium/blockentity/multiblock/MachineCoreBlockEntity.java

```diff
@@ -0,0 +1,123 @@
+package com.techarium.techarium.blockentity.multiblock;
+
+import com.techarium.techarium.Techarium;
+import com.techarium.techarium.block.multiblock.MachineCoreBlock;
+import com.techarium.techarium.inventory.MachineCoreMenu;
+import com.techarium.techarium.multiblock.MultiBlockStructure;
+import com.techarium.techarium.platform.CommonServices;
+import com.techarium.techarium.registry.TechariumBlockEntities;
+import net.minecraft.core.BlockPos;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.MenuProvider;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A BlockEntity for the {@link MachineCoreBlock}.
+ */
+public class MachineCoreBlockEntity extends BlockEntity implements MenuProvider {
+	// We don't save the selected multiblock
+	//  1- The level isn't set when the block entity is loaded
+	//  2- I (Ketheroth) don't want to cache the resource location until I can transform it in the multiblock structure.
+
+	// TODO @anyone: 12/06/2022 change this to allow displaying for just a few seconds (see TODO.md)
+	/**
+	 * List of multiblock cores that are obstructed and can't deploy. This is used to render the obstruction overlays.
+	 */
+	public static List<BlockPos> CORE_WITH_OBSTRUCTION = new ArrayList<>();
```

Comment:
This is pretty bad, this is a static non-final mutable list that is accessed from a render event(which is another thread), data being mutated while being accessed from separate threads is very unsafe.
It also appears to have a lot of `add` and `remove` operations, the latter of which is slow on lists, so I'd suggest switching this to some sort of set and not making it static, and it should be thread safe.